meta/AggregationCloud.php

```diff
@@ -0,0 +1,157 @@
+     */
+    protected $sums;
+    /**
+     * @var bool skip full table when no results found
+     */
+    protected $simplenone = true;
+    /**
+     * @todo we might be able to get rid of this helper and move this to SearchConfig
+     * @var \helper_plugin_struct_config
+     */
+    protected $helper;
+    /**
+     * Initialize the Aggregation renderer and executes the search
+     *
+     * You need to call @see render() on the resulting object.
+     *
+     * @param string $id
+     * @param string $mode
+     * @param \Doku_Renderer $renderer
+     * @param SearchConfig $searchConfig
+     */
+    public function __construct($id, $mode, \Doku_Renderer $renderer, SearchCloud $searchConfig) {
+        $this->id = $id;
+        $this->mode = $mode;
+        $this->renderer = $renderer;
+        $this->searchConfig = $searchConfig;
+        $this->data = $searchConfig->getConf();
+        $this->columns = $searchConfig->getColumns();
+        $this->result = $this->searchConfig->execute();
+        $this->resultCount = $this->searchConfig->getCount();
+        $this->helper = plugin_load('helper', 'struct_config');
+
+        $this->max = $this->result[0]['count'];
+        $this->min = end($this->result)['count'];
+    }
+
+    /**
+     * Create the table on the renderer
+     */
+    public function render() {
+        $this->startScope();
+        $this->renderer->doc .= '<ul>';
+        foreach ($this->result as $result) {
+            $this->renderResult($result);
+        }
+        $this->renderer->doc .= '</ul>';
+        $this->finishScope();
+        return;
+    }
+    /**
+     * Adds additional info to document and renderer in XHTML mode
+     *
+     * @see finishScope()
+     */
+    protected function startScope() {
+        // unique identifier for this aggregation
+        $this->renderer->info['struct_cloud_hash'] = md5(var_export($this->data, true));
+        // wrapping div
+        if($this->mode != 'xhtml') return;
+        $this->renderer->doc .= "<div class=\"structaggregation cloudaggregation\">";
+    }
+    /**
+     * Closes the table and anything opened in startScope()
+     *
+     * @see startScope()
+     */
+    protected function finishScope() {
+        // remove identifier from renderer again
+        if(isset($this->renderer->info['struct_cloud_hash'])) {
+            unset($this->renderer->info['struct_cloud_hash']);
+        }
+        // wrapping div
+        if($this->mode != 'xhtml') return;
+        $this->renderer->doc .= '</div>';
+    }
+
+    protected function renderResult($result) {
+        /**
+         * @var Value $value
+         */
+        $value = $result['tag'];
+        $count = $result['count'];
+        $weight = $this->getWeight($count, $this->min, $this->max);
+        $type = strtolower($value->getColumn()->getType()->getClass());
+        if ($value->isEmpty()) {
+            return;
+        }
+
+        $raw = $value->getRawValue();
```

Comment:
Mh, true. However there is the edge-case that display-value is empty for page-types where the page never existed or raw-value is not a valid page-id (for example because the type was changed from `Text` to `Page`).

Moreover I would actually like to render _some_ of the types, for example colors and images. `#12a6b3` is just not that useful as a tag.